src/main/scala/com/ironcorelabs/DeviceCreateOpts.scala

```diff
@@ -0,0 +1,22 @@
+package com.ironcorelabs.scala.sdk
+
+import cats.effect.Sync
+import cats.implicits._
+import com.ironcorelabs.{sdk => jsdk}
+
+/**
+ * Optional parameters for creating a new device instance.
+ */
+case class DeviceCreateOpts(
```

Comment:
I can't make a roundtrip test because there are no getter functions on the java DeviceCreateOpts. But I fixed the NullPointerException and added a test that shows that ```java:DeviceCreateOpts(null) -> scala:DeviceCreateOpts(None)```